File: backend/src/main/java/com/se104/backend/entity/Submission.java
package com.se104.backend.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "SUBMISSION")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Submission {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "submission_id")
    private Integer submissionId;

    // CHECK: score BETWEEN 0 AND 10  (enforce ở tầng app)
    @Column(name = "score", nullable = false)
    private Float score;

    @Column(name = "score_text")
    private String scoreText;

    @Column(name = "note")
    private String note;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id", nullable = false)
    private Student student;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "class_id", nullable = false)
    private Clazz clazz;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "exam_id", nullable = false)
    private Exam exam;
}

File: frontend/src/models/submission.js
export function mapSubmissionListItem(item={})
{
    return {
        submissionId: item.submissionId ?? item.id ?? null,
        studentId: item.studentId ?? null,
        // Danh sách trả về fullName, chi tiết trả về studentName. Map cả 2 về fullName.
        fullName: item.fullName ?? item.studentName ?? item.name ?? '',
        score: item.score ?? -1,
        status: item.status === true || item.status === 'true',
    };
}
export function mapSubmissionDetailItem(item={})
{
    return {
        submissionId: item.submissionId ?? item.id ?? null,
        studentId: item.studentId ?? null,
        fullName: item.studentName ?? item.fullName ?? '', // Chuẩn hóa studentName từ API về fullName
        classId: item.classId ?? null,
        score: item.score ?? -1,
        scoreText: item.scoreText ?? '',
        status: item.status === true,
        note: item.note ?? '',
    };
}
export function buildSubmissionCreateAndUpdateRequest({ examId, score, scoreText, note, status }) {
    return {
        examId,
        score,
        scoreText,
        note,
        status
    };
}